src/Raven.Pal/src/posix/mapping.c

```diff
@@ -77,21 +86,25 @@ rvn_create_and_mmap64_file(
     if (mfh->path == NULL)
     {
         rc = FAIL_CALLOC;
-        goto error_cleanup;
+        goto error_clean_with_error;
     }
     mfh->flags = flags;
-    mfh->fd = open(path, O_RDWR | O_CREAT, S_IWUSR | S_IRUSR);
+    int32_t largefile = 0;
+     if (sizeof(int) == 4) /* 32 bits */
```

Comment:
This isn't correct. `sizeof(int)` is _always_ 4 bytes on Linux and Windows.
You need to check `sizeof(void*)` if you want point size.

Reply:
Done.